heat/core/dndarray.py

```diff
@@ -568,11 +576,21 @@ def cpu(self) -> DNDarray:
         self.__device = devices.cpu
         return self
 
-    def create_lshape_map(self) -> torch.Tensor:
+    def create_lshape_map(self, recreate: Optional[bool] = True) -> torch.Tensor:
```

Comment:
I might be wrong but I seem to remember  `Optional` should only be used when the default value is `None`.

should we replace `recreate` with `force-check`?

Reply:
i removed the optional there. you are correct

i dont view it as a check per-se. it doesnt correct an already existing one, but it creates a new one instead. but i guess that is a matter of perspective. do we use `force-check` elsewhere already?

Reply:
we use `force_check` in `is_balanced()` - not sure the choice of underscore over hyphen was a sound one, in hindsight.